src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -192,14 +192,18 @@ public void Dispose()
 
         private (uint RefEscapeScope, uint ValEscapeScope) GetLocalScopes(LocalSymbol local)
         {
-            Debug.Assert(_localEscapeScopes is { });
-            return _localEscapeScopes[local];
+            Debug.Assert(_localEscapeScopes?.ContainsKey(local) == true);
+
+            return _localEscapeScopes?.TryGetValue(local, out var scopes) == true
+                ? scopes
+                : (Binder.CallingMethodScope, Binder.CallingMethodScope);
```

Comment:
The changes to `GetLocalScopes()`, `SetLocalScopes()`, and `AddLocalScopes()` is to avoid a similar potential issue with locals, if a reference to a local is visited before the local declaration.